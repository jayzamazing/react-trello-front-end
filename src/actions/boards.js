import {normalize} from 'normalizr';
import {boardArray, boardsSchema} from '../board-schema';
import {BASE_URL} from '../config';

/*
* action to tell store that all boards has been retrieved
* @params boards - boards to be sent to store
* @returns action type and boards
*/
export const FIND_BOARDS_SUCCESS = 'FIND_BOARDS_SUCCESS';
export const findBoardsSuccess = boards => {
  //grab all boards array with cards and cardslist and normalize it
  const items = (normalize(boards, boardArray)).entities;
  return {
    type: FIND_BOARDS_SUCCESS,
    items
  };
};
/*
* function to get all boards
* @params findBoardsSuccess or passed in action
* @dispatch findBoardsSuccess or passed in action
*/
export const getBoards = (action = findBoardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards`)
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(res.body));
    });
};
/*
* action to tell store that a board has been created
* @params boards - boards to be sent to store
* @returns action type and boards
*/
export const CREATE_BOARD_SUCCESS = 'CREATE_BOARD_SUCCESS';
export const createBoardSuccess = boards => {
  const items = (normalize(boards, boardsSchema)).entities;
  return {
    type: CREATE_BOARD_SUCCESS,
    items
  };
};
/*
* function to create a board
* @params postData - boards to update on the board
* @params createBoardSuccess or passed in action
* @dispatch createBoardSuccess or passed in action
*/
export const createBoards = (postData, action = createBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards`, {
    method: "POST",
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(res.body));
    });
};
/*
* action to tell store that a board has been deleted
* @params id - id of the board to delete in the store
* @returns action type and boards
*/
export const DELETE_BOARD_SUCCESS = 'DELETE_BOARD_SUCCESS';
export const deleteBoardSuccess = id => {
  return {
    type: DELETE_BOARD_SUCCESS,
    boardId: id
  };
};
/*
* function to delete a board
* @params id - id of the board to delete
* @params deleteBoardSuccess or passed in action
* @dispatch deleteBoardSuccess or passed in action
*/
export const deleteBoards = (id, action = deleteBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a board has been updated
* @params boards - boards to be sent to store
* @params id - id of the board to update in the store
* @returns action type and boards
*/
export const UPDATE_BOARD_SUCCESS = 'UPDATE_BOARD_SUCCESS';
export const updateBoardSuccess = function(id, boards) {
  const items = (normalize(boards, boardsSchema)).entities;
  return {
    type: UPDATE_BOARD_SUCCESS,
    items,
    boardId: id
  };
};
/*
* function to update the board
* @params id - id of the board to update
* @params postData - boards to update on the board
* @params updateBoardSuccess or passed in action
* @dispatch updateBoardSuccess or passed in action
*/
export const updateBoards = (id, postData, action = updateBoardSuccess) => dispatch => {
  return fetch(`${BASE_URL}/boards/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
