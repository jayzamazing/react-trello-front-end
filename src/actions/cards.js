import {normalize} from 'normalizr';
import {cardsSchema} from '../board-schema';
import {BASE_URL} from '../config';
/*
* action to tell store that a cards has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const CREATE_CARDS_SUCCESS = 'CREATE_CARDS_SUCCESS';
export const createCardsSuccess = cards => {
  const items = (normalize(cards, cardsSchema)).entities;
  return {
    type: CREATE_CARDS_SUCCESS,
    items
  };
};
/*
* function to create a cards
* @params postData - data to update on the cards
* @params createCardsSuccess or passed in action
* @dispatch createCardsSuccess or passed in action
*/
export const createCards = (postData, action = createCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards`, {
    method: "POST",
    body: postData
  })
    .then(res => {
      if (!res.ok) return Promise.reject(res.statusText);
      return dispatch(action(res.body));
    });
};
/*
* action to tell store that a cards has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const DELETE_CARDS_SUCCESS = 'DELETE_CARDS_SUCCESS';
export const deleteCardsSuccess = id => {
  return {
    type: DELETE_CARDS_SUCCESS,
    cardsId: id
  };
};
/*
* function to delete a cards
* @params id - id of the cards to delete
* @params deleteCardsSuccess or passed in action
* @dispatch deleteCardsSuccess or passed in action
*/
export const deleteCards = (id, action = deleteCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a cards has been updated
* @params cards - cards to be sent to store
* @params id - id of the cards to update in the store
* @returns action type and cards
*/
export const UPDATE_CARDS_SUCCESS = 'UPDATE_CARDS_SUCCESS';
export const updateCardsSuccess = function(id, cards) {
  const items = (normalize(cards, cardsSchema)).entities;
  return {
    type: UPDATE_CARDS_SUCCESS,
    items,
    cardsId: id
  };
};
/*
* function to update the cards
* @params id - id of the cards to update
* @params postData - cards to update on the board
* @params updateCardsSuccess or passed in action
* @dispatch updateCardsSuccess or passed in action
*/
export const updateCards = (id, postData, action = updateCardsSuccess) => dispatch => {
  return fetch(`${BASE_URL}/cards/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
