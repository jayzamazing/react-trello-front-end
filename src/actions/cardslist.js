'use strict';
import {normalize} from 'normalizr';
import {cardsListSchema} from '../board-schema';

/*
* action to tell store that a cardslist has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const CREATE_CARDSLIST_SUCCESS = 'CREATE_CARDSLIST_SUCCESS';
export const createCardslistSuccess = cardslist => {
  const items = (normalize(cardslist, cardsListSchema)).entities;
  return {
    type: CREATE_CARDSLIST_SUCCESS,
    items
  };
};
/*
* function to create a cardslist
* @params postData - data to update on the cardslist
* @params createCardslistSuccess or passed in action
* @dispatch createCardslistSuccess or passed in action
*/
export const createCardslist = (postData, action = createCardslistSuccess) => dispatch => {
  return fetch('/cardslist', {
    method: "POST",
    body: postData
  })
    .then(res => {
      if (!res.ok) return Promise.reject(res.statusText);
      return dispatch(action(res.body));
    });
};
/*
* action to tell store that a cardslist has been created
* @params data - data to be sent to store
* @returns action type and data
*/
export const DELETE_CARDSLIST_SUCCESS = 'DELETE_CARDSLIST_SUCCESS';
export const deleteCardslistSuccess = id => {
  return {
    type: DELETE_CARDSLIST_SUCCESS,
    cardslistId: id
  };
};
/*
* function to delete a cardslist
* @params id - id of the cardslist to delete
* @params deleteCardslistSuccess or passed in action
* @dispatch deleteCardslistSuccess or passed in action
*/
export const deleteCardslist = (id, action = deleteCardslistSuccess) => dispatch => {
  return fetch(`/cardslist/${id}`, {
    method: 'DELETE'
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id));
    });
};
/*
* action to tell store that a cardslist has been updated
* @params cardslist - cardslist to be sent to store
* @params id - id of the cardslist to update in the store
* @returns action type and cardslist
*/
export const UPDATE_CARDSLIST_SUCCESS = 'UPDATE_CARDSLIST_SUCCESS';
export const updateCardslistSuccess = function(id, cardslist) {
  const items = (normalize(cardslist, cardsListSchema)).entities;
  return {
    type: UPDATE_CARDSLIST_SUCCESS,
    items,
    cardslistId: id
  };
};
/*
* function to update the cardslist
* @params id - id of the cardslist to update
* @params postData - cardslist to update on the board
* @params updateCardslistSuccess or passed in action
* @dispatch updateCardslistSuccess or passed in action
*/
export const updateCardslist = (id, postData, action = updateCardslistSuccess) => dispatch => {
  return fetch(`/cardslist/${id}`, {
    method: 'PUT',
    body: postData
  })
    .then((res) => {
      if (!res.ok) return Promise.reject(res.statusText);
      dispatch(action(id, postData));
    });
};
