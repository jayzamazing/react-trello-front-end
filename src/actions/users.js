import {SubmissionError} from 'redux-form';
import {BASE_URL} from '../config';
import {normalizeResponseErrors} from './utils';

export const createUser = user => dispatch => {
  return fetch(`${BASE_URL}/users`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json'
    },
    body: JSON.stringify(user)
  }).then(res => normalizeResponseErrors(res))
  .then(res => res.json())
  .catch(err => {
    const {reason, message, location} = err;
    return Promise.reject(
      return Promise.reject(
        new SubmissionError({
          [location]: message
        });;
      );
    );
  });
}
