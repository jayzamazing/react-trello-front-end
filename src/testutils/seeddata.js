'use strict';
import faker from 'faker';
//used to create a single board in json
export const seedBoardsSingle = (title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    title: title,
    _id: _id,
    cardsList: seedCardslistIds()
  };
};
//used to create random boards based on count
export const seedBoardsCount = (count) => {
  if (!count) {
    count = 1;
  }
  let boards = [];
  for (var i = 0; i < count; i++) {
    let _id = faker.random.alphaNumeric(20);
    boards.push({
      _id: _id,
      title: faker.random.words(),
      cardsList: seedCardslistIds(count)
    });
  }
  return boards;
}
//create board items based on input values, after normalization
export const seedBoardsInput = (ids, words, ids2) => {
  let boards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    boards[_id] = {
      _id: _id,
      title: words[i],
      cardslist: ids2[i]
    };
  }
  return boards;
}
//used to create cardslist ids for seedboards
export const seedCardslistIds = (count) => {
  let cardslistIds = [];
  for (var i = 0; i < count; i++) {
    cardslistIds.push(faker.random.alphaNumeric(20));
  }
  return cardslistIds;
}
//used to create a single cardslist in json
export const seedCardslistSingle = (boardId = faker.random.alphaNumeric(20), title = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    boardId: boardId,
    title: title,
    _id: _id,
    cards: seedCardsIds()
  };
}
//used to create random cardslist based on count
export const seedCardslistCount = (boardId, count) => {
  if (!count) {
    count = 1;
  }
  let cardslist = [];
  for (var i = 0; i < count; i++) {
    cardslist.push({
      boardId: boardId,
      _id: faker.random.alphaNumeric(20),
      title: faker.random.words(),
      cards: seedCardsIds(count)
    });
  }
  return cardslist;
}
//create cardslist items based on input value, after normalization
export const seedCardslistInput = function(boardId, words, ids) {
  let cardslist = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = ids[i];
    cardslist[_id]= {
      boardId: boardId[i],
      _id: _id,
      title: words[i],
      cards: seedCardsIds(ids.length)
    };
  }
  return cardslist;
};
//used to create cardslist ids for seedboards
export const seedCardsIds = (count) => {
  let cardsIds = [];
  for (var i = 0; i < count; i++) {
    cardsIds.push(faker.random.alphaNumeric(20));
  }
  return cardsIds;
}
//used to create a single card in json
export const seedCardsSingle = (cardslistId = faker.random.alphaNumeric(20), text = faker.random.words(), _id = faker.random.alphaNumeric(20)) => {
  return {
    cardslistId: cardslistId,
    text: text,
    _id: _id
  };
}
//used to create random cardslist based on count
export const seedCardsCount = (cardslistId, count) => {
  if (!count) {
    count = 1;
  }
  let cards = [];
  for (var i = 0; i < count; i++) {
    cards.push({
      cardslistId: cardslistId,
      _id: faker.random.alphaNumeric(20),
      text: faker.random.words()
    });
  }
  return cards;
}
//create cards items based on input value, after normalization
export const seedCardsInput = function(cardslistId, words, ids) {
  let cards = {};
  for (var i = 0; i < ids.length; i++) {
    var _id = faker.random.alphaNumeric(20);
    cards[_id]= {
      cardslistId: cardslistId[i],
      _id: _id,
      text: faker.random.words()
    };
  }
  return cards;
};
//create a random title
export const createTitle = () => {
return {title: faker.random.words()};
};
