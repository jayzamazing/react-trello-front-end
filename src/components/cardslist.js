'use strict';
import React from 'react';
import Cards from './cards';
import {connect} from 'react-redux';
import * as actions from '../actions/cardslist';
import CreateItems from './create-items';
import {Immutable} from 'seamless-immutable';

//function to render multiple lists of cards
export class Cardslist extends React.Component {
  //set up initial data state
  constructor(props) {
    super(props);
    this.state = {
      showCreateCardslist: false,
      editCardslist: {},
      cardslist: {},
      cardslistTitle: ''
    }
  }
  //keep track of text
  onAddInputChanged(event) {
    //if the addCardslist input is being used
    if (event.target.name == 'addCardslist') {
      this.setState({cardslistTitle: event.target.value});
      //otherwise assume we are editing cardslist name
    } else {
      //get cardslist from state
      var temp = this.state.cardslist;
      //update the title for the selected cardslist
      var temp2 = Immutable.update(temp, event.target.id, function() {
        return {title: event.target.value};
      });
      //store the updated cardslist title
      this.setState({cardslist: temp2});
    }
  }
  //function to add a new cardslist
  addCardslist() {
    // this.props.dispatch(
    // //dispatch query boards
    // actions.createCardslist({
    //   title: this.state.cardslistTitle,
    //   boardId: this.props.params.boardId.replace(':', '')
    // }));
    //hide the following input
    this.setState({showCreateCardslist: false});
  }
  //function to delete a cardslist
  // deleteCardslist(cardslistId) {
  //   this.props.dispatch(
  //   //dispatch query cardslist
  //   actions.deleteCardslist(cardslistId));
  // }
  //function to edit the name of the cardslist
  updateCardslist(cardslistId, cardslistName) {
    this.props.dispatch(actions.updateCardslist(cardslistId, {
      title: cardslistName,
      boardId: this.props.params.boardId.replace(':', '')
    }));
    this.forceUpdate();
  }
  //set the variable to show the create cardslist inputs
  showCreateCardslist() {
    this.setState({showCreateCardslist: true});
  }
  //deal with the user hitting enter from the input and updating the cardslist
  handleKeyPress(events) {
    if (events.charCode == 13) {
      var temp = this.state.editCardslist;
      temp[events.target.id] = true;
      this.setState({editCardslist: temp});
      this.updateCardslist(events.target.id, events.target.value);
    }
  }
  //set variable to enable the editing of the cardslist name
  editCardslistName(item) {
    var temp = this.state.editCardslist;
    temp[item] = false;
    this.setState({editCardslist: temp});
  }
  render() {
    if (this.props.cardslist) {
      var boardName = this.props.params.boardName.replace(':', '');
      var boardId = this.props.params.boardId.replace(':', '');
      //iterate over props.boards and get the item that matches the boardid
      var board = this.props.boards[Object.keys(this.props.boards).find(item => {
          //if the id of props.boards matches boardid
        return this.props.boards[item]._id == boardId;
      })];
      //function to render multiple cardslist
      var cardslist = Object.keys(this.props.cardslist).map((item, index) => {
        if (board.cardslist.indexOf(this.props.cardslist[item]._id) > -1) {
          var temp = this.props.cardslist[item];
          return (
            <li key={index}>
              <input type="text" id={temp._id} value={this.state.cardslist[temp._id]
                ? this.state.cardslist[temp._id].title
                : temp.title} disabled={(this.state.editCardslist[temp._id] == undefined)
                ? true
                : this.state.editCardslist[temp._id]} onChange={this.onAddInputChanged} onKeyPress={this.handleKeyPress}/>
              <Cards cardslistId={item} key={index} boardId={boardId}/>
              <input type="button" value="Delete Cardslist" onClick={() => this.props.deleteCardslist(temp._id)} name="deleteCardslist"/>
              <input type="button" value="Edit Cardslist" onClick={this.editCardslistName.bind(null, temp._id)} name="editCardslist"/>
            </li>
          );
        }
      });
    }
    return (
      <div className="board">
        <div className="board-name">
          <h1>{boardName}</h1>
        </div>
        <div className="board-list">
          <ul>
            {cardslist}
          </ul>
          <input type="button" value="Add Cards list" onClick={() => this.showCreateCardslist()} name="addCardslist"/> {this.state.showCreateCardslist
            ? <CreateItems onAddInputChanged={(evt) => this.onAddInputChanged(evt)} addItems={() => {this.addCardslist(); this.props.addCardslist();}} name="cardslistInput"/>
            : null}
        </div>
      </div>
    );
  }

};
//allows subcription to redux updates and access to data stored in redux store
const mapStateToProps = (state) => ({
  boards: state.boards,
  cardslist: state.cardslist
});
const mapDispatchToProps = (dispatch, props) => ({
  //dispatch to delete a cardslist
  deleteCardslist: (cardslistId) => {
    dispatch(actions.deleteCardslist(cardslistId));
  },
  //dispatch to add a new cardslist
  addCardslist: () => {
    actions.createCardslist({
      title: props.cardslistTitle,
      boardId: props.params.boardId.replace(':', '')
    });
  },
  //dispatch update to cardslist name if enter key is pressed
  updateCardslist: () => {
    if(events.charCode==13) {
      dispatch(actions.updateCardslist(events.target.id, {
        title: events.target.value,
        boardId: this.props.params.boardId.replace(':', '')
      }));
    }
  }
});
//connects component to redux store
export default connect(mapStateToProps, mapDispatchToProps)(Cardslist);
