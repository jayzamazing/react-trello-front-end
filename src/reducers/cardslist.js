import Immutable from 'seamless-immutable';
import {FIND_BOARDS_SUCCESS} from '../actions/boards';
import {CREATE_CARDSLIST_SUCCESS, DELETE_CARDSLIST_SUCCESS, UPDATE_CARDSLIST_SUCCESS} from '../actions/cardslist';

const initialRepositoryState = Immutable({});

/*
* Function called when DELETE_CARDSLIST_SUCCESS is called to deal with deleting a Cardslist and removing it from state
* params state- old state before merge
* params action.cardslistid- Cardslist to delete
*/
export const deleteCardslist = (state, {cardslistId}) => state.without(cardslistId);

/*
* Function called when UPDATE_CARDSLIST_SUCCESS is called to update cardslsit with a new cardslist
* params state- old state before merge
* params action.items.cardslsit- cardslist to update
*/
export const updateCardslist = (state, {cardslist}) => state.merge(cardslist, {deep: true});

/*
* Function to deal with using various reducer functions
* params state- old state before merge
* params action- action with data to modify state
*/
export default (state = initialRepositoryState, action) => {
  switch (action.type) {
  case FIND_BOARDS_SUCCESS:
  case CREATE_CARDSLIST_SUCCESS:
  case UPDATE_CARDSLIST_SUCCESS:
    if (action.cardslist) {
      return updateCardslist(state, action);
    } else {
      return state;
    }
  case DELETE_CARDSLIST_SUCCESS:
    return deleteCardslist(state, action);
  default:
    return state;
  }
};
