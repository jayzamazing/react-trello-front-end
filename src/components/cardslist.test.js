'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow, mount} from 'enzyme';
import component, {Cardslist} from './cardslist';
import {seedCardslistInput, seedBoardsInput, seedCardsInput} from '../testutils/seeddata';
import * as actions from '../actions/cardslist';

describe('Cardslist component', () => {
  let boards = {}, cardslist = {}, cards = {}, params = {}, boardKeys, cardslistKeys, cardsKeys;
beforeEach(() => {
  let ids = ['46mdujckqs45tkdsh2jdxz', '346rfdghsdhteu021ato', 'kd5o3ve2858qggfqw4rtfda'];
  let words = ['XML Health', 'Bedfordshire Illinois Liaison', 'Sports Analyst Cambridgeshire'];
  let ids2 = ['2skdde39hzdo08aqcdgd', '14w43d4opubteu021ato', 'kd5o3ve2858vpjxd7wjp'];
  let words2 = ['XML Health', 'Bedfordshire Illinois Liaison', 'Sports Analyst Cambridgeshire'];
  let ids3 = ['432e39hzdo08aqcdgd', '6532opubteu021ato', '5643e2858vpjxd7wjp'];
  let words3 = ['eat dinner', 'tea tree oil', 'masking tape'];
  boards = seedBoardsInput(ids, words, ids2);
  cardslist = seedCardslistInput(ids, words2, ids2);
  cards = seedCardsInput(ids2, words3, ids3);
  boardKeys = Object.keys(boards);
  cardslistKeys = Object.keys(cardslist);
  cardsKeys = Object.keys(cards);
  params = {
    boardName: ':' + boards[boardKeys[0]].title,
    boardId: ':' + boardKeys[0]
  };
});
  afterEach(() => {
    boards = {}, cardslist = {}, cards = {}, params = {};
  });
  //ensure cardslist renders without any issues
  it('should render', () => {
    const dispatch = jest.fn();
    shallow(<Cardslist params={params} boards={boards} cardslist={cardslist} cards={cards} dispatch={dispatch}/>)
  });
  //render and check against previous snapshot
  it('cardslist snapshot', () => {
    const dispatch = jest.fn();
    const wrapper = shallow(<Cardslist params={params} boards={boards} cardslist={cardslist} cards={cards} dispatch={dispatch}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on add button and check before and after snapshots
  it('add cardslist click snaphot', () => {
    const dispatch = jest.fn();
    const wrapper = shallow(<Cardslist params={params} boards={boards} cardslist={cardslist} cards={cards} dispatch={dispatch}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('[name="addCardslist"]').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //check dispatch is fired off when delete button is clicked
  it('dispatches delete cardslist', () => {
    const deleteCardslist = jest.fn();
    // const dispatch = jest.fn();
    const wrapper = shallow(<Cardslist params={params} boards={boards} cardslist={cardslist} cards={cards} deleteCardslist={deleteCardslist}/>);
    wrapper.find('[name="deleteCardslist"]').at(0).simulate('click');
    expect(deleteCardslist).toHaveBeenCalledWith(cardslistKeys[0]);
  });
  //click on delete button and check before and after snapshots
  it('delete cardslist snapshot', () => {
    const deleteCardslist = jest.fn();
    // const dispatch = jest.fn();
    const wrapper = shallow(<Cardslist params={params} boards={boards} cardslist={cardslist} cards={cards} deleteCardslist={deleteCardslist}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('[name="deleteCardslist"]').at(0).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //check dispatch is fired when update button is clicked
  it('dispatches update cardslist', () => {
    const updateCardslist = jest.fn();
    const wrapper = shallow(<Cardslist params={params} boards={boards} cardslist={cardslist} cards={cards} updateCardslist={updateCardslist}/>);
    wrapper.find('[name="cardslistName"]').at(0).simulate('keypress', {key: 'Enter'});
    expect(updateCardslist).toHaveBeenCalledWith({"key": "Enter"});
  });
  //click on update button and check before and after snapshots
  it('update cardslist snapshot', () => {
    const updateCardslist = jest.fn();
    const wrapper = shallow(<Cardslist params={params} boards={boards} cardslist={cardslist} cards={cards} updateCardslist={updateCardslist}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('[name="editCardslist"]').at(0).simulate('click');
    const keys = Object.keys(boards);
    wrapper.find('[name="cardslistName"]').at(0).simulate('change', { target: { value: 'testing', id: keys[0] } });
    wrapper.find('[name="cardslistName"]').at(0).simulate('keypress', {key: 'Enter'})
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
