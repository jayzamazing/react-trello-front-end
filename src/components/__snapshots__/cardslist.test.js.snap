// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cardslist component add cardslist click snaphot 1`] = `
<div
  className="board"
>
  <div
    className="board-name"
  >
    <h1>
      XML Health
    </h1>
  </div>
  <div
    className="board-list"
  >
    <ul>
      <li
        key="0"
      >
        <input
          disabled={true}
          id="2skdde39hzdo08aqcdgd"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="XML Health"
        />
        <Connect(Cards)
          boardId="46mdujckqs45tkdsh2jdxz"
          cardslistId="2skdde39hzdo08aqcdgd"
          key="0"
        />
        <input
          name="deleteCardslist"
          onClick={[Function]}
          type="button"
          value="Delete Cardslist"
        />
        <input
          name="editCardslist"
          onClick={[Function]}
          type="button"
          value="Edit Cardslist"
        />
      </li>
    </ul>
    <input
      name="addCardslist"
      onClick={[Function]}
      type="button"
      value="Add Cards list"
    />
     
  </div>
</div>
`;

exports[`Cardslist component add cardslist click snaphot 2`] = `
<div
  className="board"
>
  <div
    className="board-name"
  >
    <h1>
      XML Health
    </h1>
  </div>
  <div
    className="board-list"
  >
    <ul>
      <li
        key="0"
      >
        <input
          disabled={true}
          id="2skdde39hzdo08aqcdgd"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="XML Health"
        />
        <Connect(Cards)
          boardId="46mdujckqs45tkdsh2jdxz"
          cardslistId="2skdde39hzdo08aqcdgd"
          key="0"
        />
        <input
          name="deleteCardslist"
          onClick={[Function]}
          type="button"
          value="Delete Cardslist"
        />
        <input
          name="editCardslist"
          onClick={[Function]}
          type="button"
          value="Edit Cardslist"
        />
      </li>
    </ul>
    <input
      name="addCardslist"
      onClick={[Function]}
      type="button"
      value="Add Cards list"
    />
     
    <CreateItems
      addItems={[Function]}
      name="cardslistInput"
      onAddInputChanged={[Function]}
    />
  </div>
</div>
`;

exports[`Cardslist component cardslist snapshot 1`] = `
<div
  className="board"
>
  <div
    className="board-name"
  >
    <h1>
      XML Health
    </h1>
  </div>
  <div
    className="board-list"
  >
    <ul>
      <li
        key="0"
      >
        <input
          disabled={true}
          id="2skdde39hzdo08aqcdgd"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="XML Health"
        />
        <Connect(Cards)
          boardId="46mdujckqs45tkdsh2jdxz"
          cardslistId="2skdde39hzdo08aqcdgd"
          key="0"
        />
        <input
          name="deleteCardslist"
          onClick={[Function]}
          type="button"
          value="Delete Cardslist"
        />
        <input
          name="editCardslist"
          onClick={[Function]}
          type="button"
          value="Edit Cardslist"
        />
      </li>
    </ul>
    <input
      name="addCardslist"
      onClick={[Function]}
      type="button"
      value="Add Cards list"
    />
     
  </div>
</div>
`;
